### ISSUE TITLE
Run Migrate and Seeder on new domain

### ISSUE THREAD

**Issue by IamJustAKid** [gecche/laravel-multidomain#7]
Hi there, 

I am really glad you created this package. It mostly solves all of my problems. But, let's say, I created a new domain tenant1.site.com. 

How do I run the migrations and seeders on this tenant1 database?

I would really appreciate it If you could add some command to migrate to a new domain database.

like: php artisan domain:migrate tenant1

Thanks & Regards
Bilal Younas

**Comment by gecche** [comment]
Hi,

to migrate and seed a new tenant1.site.com domain, you just put the right
database in the .env.tenant1.site.com file and simply run the commands with
the --domain option:

php artisan migrate --domain=tenant1.site.com

Let me know

Thank you

Giacomo

Il giorno ven 8 nov 2019 alle ore 06:24 Bilal Younas <
notifications@github.com> ha scritto:

**Comment by IamJustAKid** [comment]
![image](https://user-images.githubusercontent.com/20171838/68497563-a6705e80-0276-11ea-99e9-467b3a8705be.png)

I ran the command, It only created the migration table but did not run the rest of the migrations

**Comment by IamJustAKid** [comment]
The same thing happened with key:generate command, the key was generated in default .env file but NOT in .env.tenant1.ims.local

**Comment by gecche** [comment]
Hi,

I need to see some more details about your code or configuration because
all tests I ran worked perfectly fine with all artisan commands.
Maybe you set a certain configuration in your project which causes some
issues but I need to see the details.

Cheers

Giacomo

Il giorno ven 8 nov 2019 alle ore 18:38 Bilal Younas <
notifications@github.com> ha scritto:

**Comment by IamJustAKid** [comment]
Hi there,
I am using Laravel Modules. Maybe that is creating some issues. I will check in detail and let you know. Sorry for misleading the issue.

https://nwidart.com/laravel-modules/

It will take a couple of days.

Thanks

**Comment by gecche** [comment]
Ok, I never tried that package, maybe there are some issues... but, just to run some more tests: which version of Laravel are you using?

Giacomo

Inviato da iPad

**Comment by IamJustAKid** [comment]
latest 6.2

**Comment by IamJustAKid** [comment]
When I add this package along with Laravel Module Package, Laravel Module Package stops working

I am guessing because of some kernel and provider changes

**Comment by gecche** [comment]
ok, if the laravel-module package changes the kernels there could be some issues in installing both.

However, when I have some time, I will try that package which looks interesting and I will see if something could be addressed easily.

I will let you know asap, in the meanwhile I close the issue

thanks

giacomo


Inviato da iPhone

**Comment by gecche** [comment]
Hi,

I've just have had a hour of time to check out the laravel-modules package
very quickly.

I've prepared a fresh installation of the latest Laravel 6 (6.5.0).
In composer.json I only added:

"gecche/laravel-multidomain": "~2.0",
"nwidart/laravel-modules": "^6.0"


in the require section.

I then followed the installation instructions of both modules and I added a
Blog module as described in their website.

Then I added two domains, namely site1.test and site2.test via:

php artisan domain:add site1.test
php artisan domain:add site2.test

Then I specified two distinct dbs in the .env.site1.test and
.env.site2.test and I tried to run the key:generate and the migrate
commands on both domains.

Results are as expected: the keys have been changed and the migrations have
been ran correctly on the two dbs.
Please only note that the key:generate command does not have any effect if
within the .env file there is no APP_KEY key. This is the standard behavior
of the command. One time it happened to me and I wondered why it didn't
work. So, please check if your generated .env.<DOMAIN> file has the APP_KEY
key in it.

The output of the commands are as follows:
[image: homestead.png]

So, let me know if you can replicate that test or if you have some other
stuff which could cause the issue.

Cheers,

Giacomo





Il giorno lun 11 nov 2019 alle ore 12:54 Giacomo Terreni <
giacomo.terreni@gmail.com> ha scritto:

**Comment by IamJustAKid** [comment]
To generate a key, you need to have an empty APP_KEY= in the .env.site.com

which is not added automatically, when you use domain:add

I added it separately in .env.site.com and it worked.

**Comment by IamJustAKid** [comment]
![image](https://user-images.githubusercontent.com/20171838/68649180-dbfd9c00-0543-11ea-90ad-3490a4164003.png)

but its still not running the migrations in different modules

**Comment by IamJustAKid** [comment]
and its also not recognizing each modules Service Providers and whats is in them like view composer variables

**Comment by IamJustAKid** [comment]
I have like three modules at the start and each of these modules have migrations in 

/Modules/UserManagement/Database/Migrations

and its Service Providers but its not recognizing the Service Provider of each module. 

if it starts to recognize the service providers, everything will resolve

**Comment by IamJustAKid** [comment]
OK before auto-discovery feature was working on my module provides but after adding this package somehow that stopped, 

so adding all of my modules providers in app.php providers array did the trick

**Comment by IamJustAKid** [comment]
Now, Its migrating, recognizing the providers etc.

**Comment by gecche** [comment]
Hi,

> Now, Its migrating, recognizing the providers etc.
>
great! I'm glad that it works now even with a manual configuration.

Anyway I'm a bit lost... are you saying that the multidomain package
somehow breaks the discovery feature of the laravel-modules package or I
didn't understand? In the first case I really can't figure it out why and I
will try to check it when I can.

Instead, the fact that you reported:

To generate a key, you need to have an empty APP_KEY= in the .env.site.com

which is not added automatically, when you use domain:add

I added it separately in .env.site.com and it worked.

It could be a minor issue of the domain:add command. It shouldn't skip the
APP_KEY line if it is in the original .env. I will investigate asap.

**Comment by gecche** [comment]
Hi again,

I just updated the package and the issue about the missing APP_KEY int he generated envs seems to be solved.
This was due to the presence of the character "=" in the value of the original .env: lines with such a character in their values were wrongly skipped.

The potential interference with the laravel-modules package is still under investigation but I probably need a real-case scenario to check it out properly and it could take some time.

I close the issue

Cheers

Giacomo

**Comment by IamJustAKid** [comment]
Hi there,

Yes, I have about three basic modules User Management, Setting, Activity Logging. User Management and Setting are publishing some variables globally to be accessed all out the front end. 

Before installing this package every provider was being auto-discovered. But when I installed this package and set it up. I got an error. Migrations were not being located as well as other configurations.

SO I added these Service Providers manually to app.php and ALL IS GOOD NOW.
